qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionWithSwitchWhenSectionTest.java

```diff
@@ -0,0 +1,178 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.completions;
+
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for Qute completion in expression for #switch and #when section.
+ *
+ */
+public class QuteCompletionInExpressionWithSwitchWhenSectionTest {
+
+	@Test
+	public void propertyPartWhenNoExisting() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is | }\r\n" + // <-- completion here
+				"{/when}";
+		testCompletionFor(template, //
+				c("ON : MachineStatus", "ON", r(2, 7, 2, 7)), //
+				c("OFF : MachineStatus", "OFF", r(2, 7, 2, 7)), //
+				c("BROKEN : MachineStatus", "BROKEN", r(2, 7, 2, 7)));
+	}
+	
+	@Test
+	public void propertyPartSwitchIsNoExisting() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#switch Machine.status}\r\n" + //
+				"  {#is | }\r\n" + // <-- completion here
+				"{/switch}";
+		testCompletionFor(template, //
+				c("ON : MachineStatus", "ON", r(2, 7, 2, 7)), //
+				c("OFF : MachineStatus", "OFF", r(2, 7, 2, 7)), //
+				c("BROKEN : MachineStatus", "BROKEN", r(2, 7, 2, 7)));
+	}
+
+	@Test
+	public void propertyPartWhenAlreadyCompleted() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is ON |}\r\n" + // <-- should be no completion here with no in keyword
+				"{/when}";
+		testCompletionFor(template);
+	}
+	
+	@Test
+	public void propertyPartWhenAlreadyCompletedNotOperator() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is not ON |}\r\n" + // <-- should be no completion here with not as keyword
+				"{/when}";
+		testCompletionFor(template);
+	}
```

Comment:
Replace with

testCompletionFor(template, 0);

Im not sure this test works.
Indeed  I dont see the code which manage that. I suggest that you use 
https://github.com/redhat-developer/quarkus-ls/blob/c76ddfb1219cef1cb31206c023babfa888e60c76/qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/QuteAssert.java#L123 for all your tests. I mean set the expectzd count and all expected  comoletion items.